samza-azure/src/test/java/org/apache/samza/system/azureblob/avro/TestAzureBlobOutputStream.java

```diff
@@ -0,0 +1,347 @@
+    azureBlobOutputStream.close();
+
+    verify(mockCompression).compress(bytes);
+    ArgumentCaptor<Flux> argument = ArgumentCaptor.forClass(Flux.class);
+    verify(mockBlobAsyncClient).stageBlock(eq(blockIdEncoded(0)), argument.capture(), eq((long) compressedBytes.length)); // since size of byte[] written is less than threshold
+    Assert.assertEquals(ByteBuffer.wrap(compressedBytes), ((Flux) argument.getValue()).blockFirst());
+    verify(mockMetrics, times(2)).updateWriteByteMetrics(THRESHOLD / 2);
+    verify(mockMetrics, times(1)).updateAzureUploadMetrics();
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void testWriteFailed() {
+    when(mockBlobAsyncClient.stageBlock(anyString(), any(), anyLong())).thenThrow(new Exception("Test Failed"));
+
+    byte[] b = new byte[100];
+    azureBlobOutputStream.write(b, 0, THRESHOLD); // threshold crossed so stageBlock is scheduled.
+    // azureBlobOutputStream.close waits on the CompletableFuture which does the actual stageBlock in uploadBlockAsync
+    azureBlobOutputStream.close();
+  }
+
+  @Test
+  public void testClose() {
+    azureBlobOutputStream.write(bytes, 0, THRESHOLD);
+    int blockNum = 0;
+    String blockId = String.format("%05d", blockNum);
+    String blockIdEncoded = Base64.getEncoder().encodeToString(blockId.getBytes());
+
+    PowerMockito.doAnswer(invocation -> {
+        ArrayList<String> blockListArg = (ArrayList<String>) invocation.getArguments()[0];
+        String blockIdArg = (String) blockListArg.toArray()[0];
+        Assert.assertEquals(blockIdEncoded, blockIdArg);
+        Map<String, String> blobMetadata = (Map<String, String>) invocation.getArguments()[2];
+        Assert.assertEquals(blobMetadata.get(AzureBlobOutputStream.BLOB_RAW_SIZE_BYTES_METADATA), Long.toString(THRESHOLD));
+        return Mono.just(new SimpleResponse(null, 200, null, null));
+      }).when(mockBlobAsyncClient).commitBlockListWithResponse(anyList(), any(), any(), any(), any());
+
+    azureBlobOutputStream.close();
+    verify(mockMetrics).updateAzureCommitMetrics();
+  }
+
+  @Test
+  public void testCloseMultipleBlocks() {
+    azureBlobOutputStream.write(bytes, 0, THRESHOLD);
+    azureBlobOutputStream.write(bytes, 0, THRESHOLD);
+
+    int blockNum = 0;
+    String blockId = String.format("%05d", blockNum);
+    String blockIdEncoded = Base64.getEncoder().encodeToString(blockId.getBytes());
+
+    int blockNum1 = 1;
+    String blockId1 = String.format("%05d", blockNum1);
+    String blockIdEncoded1 = Base64.getEncoder().encodeToString(blockId1.getBytes());
+
+    PowerMockito.doAnswer(invocation -> {
+        ArrayList<String> blockListArg = (ArrayList<String>) invocation.getArguments()[0];
+        String blockIdArg = (String) blockListArg.toArray()[0];
+        String blockIdArg1 = (String) blockListArg.toArray()[1];
+        Assert.assertEquals(blockIdEncoded, blockIdArg);
+        Assert.assertEquals(blockIdEncoded1, blockIdArg1);
```

Comment:
Can this be replaced with `Assert.assertEquals(Arrays.asList(blockIdEncoded, blockIdEncoded1), blockListArg)`?